File: problem/ZJYYCOJ/250205/M.cpp
#include <bits/stdc++.h>
using namespace std;
const int N = 1e5+10;
int n,arr[N],d;
int main(){
    int ans = 0;
    cin>>n;
    for(int i = 1;i<=n;i++){
        cin>>arr[i];
    }
    for(int i = 1;i<=n;i++){
        if(arr[i]>arr[i-1])ans = arr[i];
        else break;
    }
    cout<<ans;
    return 0;
}

File: problem/ZJYYCOJ/C250319/M.py
##还得是python，暴力过数论题
k = int(input())
arr = [7 % k]
[arr.append((arr[-1] * 10 + 7) % k) for i in range(10**6)]
ans = -1

for i, a in enumerate(arr):
    if a == 0:
        ans = i + 1
        break

print(ans)